src/syscheckd/run_realtime.c

```diff
@@ -91,6 +91,7 @@ int realtime_adddir(const char *dir, __attribute__((unused)) int whodata)
 
                 if (!OSHash_Get_ex(syscheck.realtime->dirtb, wdchar)) {
                     if (retval = OSHash_Add_ex(syscheck.realtime->dirtb, wdchar, data), retval == 0) {
+                        os_free(data);
```

Comment:
If this code means that `data` may remain dangling if `OSHash_Add_ex()` returns `0` (error), we should also contemplate that `OSHash_Add_ex()` returned `1` (duplicate entry).

I suggest:

```c
if (retval = OSHash_Add_ex(...), retval != 2) {
   ...
}
```

Reply:
I review it and data is free
```C
if (retval = OSHash_Add_ex(syscheck.realtime->dirtb, wdchar, data), retval == 0) {
     os_free(data);
     merror_exit(FIM_CRITICAL_ERROR_OUT_MEM);
 } else if (retval == 1) {
     mdebug2(FIM_REALTIME_HASH_DUP, data);
     os_free(data);
 }
mdebug1(FIM_REALTIME_NEWDIRECTORY, dir);
```